tests/ttnn/unit_tests/operations/test_all_reduce.py

```diff
@@ -0,0 +1,234 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+import pytest
+from loguru import logger
+import ttnn
+from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import comp_equal, comp_pcc
+from models.utility_functions import skip_for_grayskull, get_devices_for_t3000
+
+
+def is_unsupported_case(input_shape, math_op, mem_config, num_devices, num_links, input_dtype, layout):
+    elem_size = 2 if input_dtype == ttnn.bfloat16 else 1
+    tensor_size_bytes = elem_size
+    for i in input_shape:
+        tensor_size_bytes *= i
+    num_l1_banks = 64
+    if mem_config.buffer_type == ttnn.BufferType.L1 and tensor_size_bytes > num_l1_banks * 50 * 1024:
+        return True, "L1 buffer can't support large tensor sizes"
+
+    return False, ""
+
+
+def run_with_trace(
+    t3k_mesh_device,
+    input_tensor_mesh,
+    num_links,
+    math_op,
+    output_mem_config,
+    n_worker,
+    n_buffer,
+    num_iters,
+):
+    # Compile Run
+    logger.info("Compiling model")
+    output_tensor_mesh = ttnn.experimental.all_reduce(
+        input_tensor_mesh,
+        math_op=math_op,
+        num_links=num_links,
+        memory_config=output_mem_config,
+        num_workers=n_worker,
+        num_buffers_per_channel=n_buffer,
+    )
+    for device_id in t3k_mesh_device.get_device_ids():
+        ttnn.synchronize_device(t3k_mesh_device.get_device(device_id))
+
+    # Capture trace
+    logger.info("Capturing trace")
+    trace_id = ttnn.begin_trace_capture(t3k_mesh_device, cq_id=0)
+    for i in range(num_iters):
+        output_tensor_mesh = ttnn.experimental.all_reduce(
+            input_tensor_mesh,
+            math_op=math_op,
+            num_links=num_links,
+            memory_config=output_mem_config,
+            num_workers=n_worker,
```

Comment:
let's avoid default valuing the `num_workers` and `num_buffers_per_channel` to avoid the test diverging from the implementation. The implementation should be free to choose with support for overridden values as best effort support.